src/System.CommandLine.Hosting/HostingExtensions.cs

```diff
@@ -141,11 +141,14 @@ public static InvocationContext GetInvocationContext(this HostBuilderContext con
             throw new InvalidOperationException("Host builder has no Invocation Context registered to it.");
         }
 
-        public static IHost GetHost(this InvocationContext invocationContext)
+        public static IHost? GetHost(this InvocationContext invocationContext)
         {
             _ = invocationContext ?? throw new ArgumentNullException(paramName: nameof(invocationContext));
             var hostModelBinder = new ModelBinder<IHost>();
-            return (IHost)hostModelBinder.CreateInstance(invocationContext.BindingContext);
+            var instance = hostModelBinder.CreateInstance(invocationContext.BindingContext);
+            if (instance is IHost host)
+              return host;            
+            return null;
```

Comment:
I would think we should  not allow null here.

Reply:
I think so before, from test code <https://github.com/dotnet/command-line-api/blob/209b724a3c843253d3071e8348c353b297b0b8b5/src/System.CommandLine.Hosting.Tests/HostingTests.cs#L401-L403>
it may be null, if this should not be `null`, guess we should throw an `InvalidOperationException`?